ecs-init/config/common.go

```diff
@@ -135,6 +135,14 @@ func CacheDirectory() string {
 	return directoryPrefix + "/var/cache/ecs"
 }
 
+// UpdateDownloadDir returns to location of the agent update download dir
+func UpdateDownloadDir() string {
+	if downloadDir := os.Getenv("ECS_UPDATE_DOWNLOAD_DIR"); downloadDir != "" {
```

Comment:
for AL2, the envs seen by ecs-init process contains the ones in /etc/ecs/ecs.config because of [this line](https://github.com/aws/amazon-ecs-init/blob/master/packaging/amazon-linux-ami/ecs.service#L26). But for AL1, IIUC i don't think they can be retrieved from os.Getenv and I think you need to get the value from [here](https://github.com/aws/amazon-ecs-init/blob/master/ecs-init/docker/docker.go#L203)

Reply:
good catch, fixing